be2-scala/src/main/scala/ch/epfl/pop/model/network/method/message/data/DataBuilder.scala

```diff
@@ -33,10 +35,21 @@ object DataBuilder {
     case _ => throw new ProtocolException(s"Unknown object '${_object}' encountered while creating a Data")
   }
 
+  /**
```

Comment:
you might want to have a chat with the Android team, as it might make your development faster.
They switched to a message "registry", i.e. they have a gigantic table that contains ObjectType/ActionType/`buildFromJson` call.

This way they support ALL `buildData` calls by essentially doing (pseudocode): 
```
ref = MessageRegistry.getInfo(object)(action)
if ref == null { throw "unsupported message" }
buildOrReject(payload)(ref.schemaValdiator)(ref.buildFromJson)
```

This leads to less code and just one big data table containing everything.